### ISSUE TITLE
ZAP checks robots file before doing script base authentication when doing indexation

### ISSUE THREAD

**Issue by Wawki** [zaproxy/zaproxy#1901]
When we check the option "check robots.txt", when doing indexation, ZAP checks it before doing the script base authentication.

It's a problem for our website because when you go on it, it gives you a cookie. After the authentication, we have another cookie. The website checks for the 2 cookies before allow access to a page.

So when ZAP checks for robots.txt, the website gives him a cookie (example "ABC") and the pages discovered after that are made with this cookie, so here the user is not authenticated and we may encountered some redirect to login page.

In parallel, we have our script base authentication, here the website gives him another cookie (example "CDF") . The script authenticates the user and receives another cookie (example "12345") (here we have 2 cookies). So the pages discovered after that are made with these 2 cookies. All is good here :)

In short we have 2 authentication cookies in parallel ("ABC" and "CDF") running around because of multithreading.

I'm not sure to be clear , please get back to me if I am not.

**Comment by kingthorin** [comment]
So you're saying access to your robots.txt requires authentication?

**Comment by Wawki** [comment]
No for example I have a page http://example.com/pageA.html discovered thanks to the robots.txt

Here we are not authenticated because the analyse of robots.txt is done before the script base authentication. 

Imagine we have a link in pageA.html that link to pageB.html. The pageB need authentication before access, so here we have a 302 (redirect to the login page), that's not what we want.

Meanwhile, script base authentication is done and some other pages are discovered (for example pageC.html)

Finally, we'll have :

pageA.html : Ok
pageB.html : Not Ok - Redirect to login page
pageC.html : Ok

Whereas we want all the pages to be ok because we have done a script base authentication.
